crates/nu-command/src/filters/find.rs

```diff
@@ -0,0 +1,165 @@
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        let span = call.head;
+
+        let ctrlc = engine_state.ctrlc.clone();
+        let engine_state = engine_state.clone();
+
+        match call.get_flag_expr("predicate") {
+            Some(predicate) => {
+                let block_id = predicate
+                    .as_block()
+                    .ok_or_else(|| ShellError::TypeMismatch("expected block".to_owned(), span))?;
+
+                let block = engine_state.get_block(block_id).clone();
+                let var_id = block.signature.get_positional(0).and_then(|arg| arg.var_id);
+
+                let mut stack = stack.collect_captures(&block.captures);
+
+                Ok(input
+                    .into_interruptible_iter(ctrlc)
+                    .find(move |value| {
+                        if let Some(var_id) = var_id {
+                            stack.add_var(var_id, value.clone());
+                        }
+
+                        eval_block(&engine_state, &mut stack, &block, PipelineData::new(span))
+                            .map_or(false, |pipeline_data| {
+                                pipeline_data.into_value(span).is_true()
+                            })
+                    })
+                    .unwrap_or(Value::Nothing { span })
+                    .into_pipeline_data())
+            }
+            None => {
+                let terms = call.rest::<Value>(&engine_state, stack, 0)?;
+                Ok(input
+                    .into_iter()
+                    .filter(move |value| {
+                        terms.iter().any(|term| match value {
+                            Value::Bool { .. }
+                            | Value::Int { .. }
+                            | Value::Filesize { .. }
+                            | Value::Duration { .. }
+                            | Value::Date { .. }
+                            | Value::Range { .. }
+                            | Value::Float { .. }
+                            | Value::Block { .. }
+                            | Value::Nothing { .. }
+                            | Value::Error { .. } => {
+                                value.eq(span, term).map_or(false, |value| value.is_true())
+                            }
+                            Value::String { .. }
+                            | Value::List { .. }
+                            | Value::CustomValue { .. } => term
+                                .r#in(span, value)
+                                .map_or(false, |value| value.is_true()),
+                            Value::Record { vals, .. } => vals.iter().any(|val| {
+                                term.r#in(span, val).map_or(false, |value| value.is_true())
+                            }),
+                            Value::Binary { .. } => todo!(),
+                            Value::CellPath { .. } => todo!(),
```

Comment:
Not sure how those should be handled.

Reply:
I'm not sure. I'd have to look to see how nushell handles them. Was there anything useful there when you looked?

Reply:
In nushell `find` converts each element of the input stream to a trimmed, lowercase string using `convert_to_string().trim().to_lowercase()`. Then it uses `str::contains` with the rest parameters of the command.

I tried to mimic this behavior for `Value::CellPath` in engine-q (see 45c006f).
However, the `convert_to_string()` method on `Primivite::Binary` in nushell always return an empty string, so we need to decide if we choose to change it, and if so how should it be changed.

Reply:
@arthur-targaryen - I don't think we support it just yet, but if you want, I think it'd be fine to file an issue and go with the assumption that we'll add support in the future. It shouldn't directly impact the find work.